### PR TITLE
helm: add description for some Helm values

### PR DESCRIPTION
**Pull request by my-git9**
Signed-off-by: xin.li <xin.li@daocloud.io>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: Add description for some values(marked TODO) in install/kubernetes/cilim/values.yaml

### CODE REVIEW

**Comment by my-git9** [comment]
/release-note/ci

File: install/kubernetes/cilium/values.yaml [hunk 1]
```diff
@@ -431,10 +431,10 @@ enableCnpStatusUpdates: false
 # mirroring it into the KVstore for reduced overhead in large clusters.
 enableK8sEventHandover: false
 
-# -- Enable setting identity mark for local traffic (default true)
+# -- Enable setting identity mark for local traffic (default true).
```
Comment (sayboras) [inline install/kubernetes/cilium/values.yaml]: is the default value true ?

File: install/kubernetes/cilium/values.yaml [hunk 2]
```diff
@@ -1350,9 +1351,10 @@ sockops:
   # enabled enables installation of socket options acceleration.
   enabled: false
 
-# TODO: Add documentation, default value
+# -- Enable check of service source ranges (currently, only for LoadBalancer) (default true).
 # svcSourceRangeCheck:
```
Comment (sayboras) [inline install/kubernetes/cilium/values.yaml]: unrelated to this PR.

This flag is not used in helm at all, can you help to add the same into configmap ? Thanks.

File: install/kubernetes/cilium/values.yaml [hunk 3]
```diff
@@ -1350,9 +1351,10 @@ sockops:
...
 
+# -- Synchronize Kubernetes nodes to kvstore and perform CNP GC (default true).
 # synchronizeK8sNodes: true
```
Comment (sayboras) [inline install/kubernetes/cilium/values.yaml]: same as above, this flag is not used at all in helm chart.

**Review by sayboras — COMMENTED** [review]
Thanks for your PR, please find below my comments.

Also, seems like the second commit was supposed to be merged with first one.

File: install/kubernetes/cilium/values.yaml (hunk 2 above)
Comment (my-git9) [reply install/kubernetes/cilium/values.yaml]: There are relevant references in the configmap:
```
{{- if .Values.svcSourceRangeCheck }}
  enable-svc-source-range-check: {{ .Values.svcSourceRangeCheck | quote }}
{{- end }}```

 I think we just need to open the annotation and assign a value here.

File: install/kubernetes/cilium/values.yaml (hunk 3 above)
Comment (my-git9) [reply install/kubernetes/cilium/values.yaml]: Same as above, please review again, thanks

File: install/kubernetes/cilium/values.yaml (hunk 2 above)
Comment (sayboras) [reply install/kubernetes/cilium/values.yaml]: sorry about this, my helm plugin in IDE seems broken

File: install/kubernetes/cilium/values.yaml (hunk 1 above)
Comment (sayboras) [reply install/kubernetes/cilium/values.yaml]: I mean the next line was having value as false, while for other fields (e.g. svcSourceRangeCheck, synchronizeK8sNodes) are uncommented with default value. 

I think it's making sense to set have helm value with default value.

**Review by sayboras — COMMENTED** [review]
LGTM with one minor comment above

Please re-run the below targets to update docs

```
make -C Documentation update-helm-values 
make -C install/kubernetes docs
```

File: install/kubernetes/cilium/values.yaml (hunk 1 above)
Comment (sayboras) [reply install/kubernetes/cilium/values.yaml]: @my-git9 curious to know what you think on the above.

File: install/kubernetes/cilium/values.yaml [hunk 4]
```diff
@@ -1350,10 +1351,11 @@ sockops:
   # enabled enables installation of socket options acceleration.
   enabled: false
 
-# TODO: Add documentation, default value
-# svcSourceRangeCheck:
+# -- Enable check of service source ranges (currently, only for LoadBalancer) (default true).
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml]: I don't think it's necessary to provide the default value between parenthesis, here and elsewhere. In the current file, the default is right on the following line; in the generated documentation, it's picked up and displayed already, no need to have it a second time.

File: install/kubernetes/cilium/values.yaml [hunk 5]
```diff
@@ -1024,13 +1025,13 @@ hubble:
...
+# -- Time to wait before using new identity on endpoint identity change (default 5s).
 # identityChangeGracePeriod: "5s"
 
-# TODO: Add documentation
+# -- GC interval for security identities (default 15m0s).
 # identityGCInterval:
 
-# TODO: Add documentation
+# -- Timeout after which identity expires on lack of heartbeat (default 30m0s).
 # identityHeartbeatTimeout: ""
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml]: Why aren't those lines getting their default value assigned as for e.g. `identityChangeGracePeriod`, even if they're commented?

File: install/kubernetes/cilium/values.yaml (hunk 1 above)
Comment (qmonnet) [reply install/kubernetes/cilium/values.yaml]: Looks like it defaults to `true` in pkg/defaults/defaults.go.

File: install/kubernetes/cilium/values.yaml [hunk 6]
```diff
@@ -431,10 +431,11 @@ enableCnpStatusUpdates: false
 # mirroring it into the KVstore for reduced overhead in large clusters.
 enableK8sEventHandover: false
 
-# TODO: Add documentation
-# enableIdentityMark: false
+# -- Enable setting identity mark for local traffic (default true).
+# enableIdentityMark: true
 
-# enableK8sEndpointSlice: false
+# -- Enables k8s EndpointSlice feature in Cilium if the k8s cluster supports it (default true).
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml]:
```suggestion
# -- Enable Kubernetes EndpointSlice feature in Cilium if the cluster supports it (default true).
```

**Review by qmonnet — CHANGES_REQUESTED** [review]
Thank you for the contribution! Please find some comments below.

Note that I would also recommend adding a bit more context in the commit description; in particular, it would be helpful to mention where the default values are coming from, so that it's easy to find from the git history in the future.

File: install/kubernetes/cilium/values.yaml [hunk 7]
```diff
@@ -1024,14 +1025,14 @@ hubble:
 # -- Method to use for identity allocation (`crd` or `kvstore`).
 identityAllocationMode: "crd"
 
-# TODO: Add documentation
+# -- Time to wait before using new identity on endpoint identity change (default 5s).
```
Comment (qmonnet) [inline install/kubernetes/cilium/values.yaml]: Any particular reason for keeping the default values in the comments for commented options?

**Review by qmonnet — APPROVED** [review]
Thanks!

**Review by sayboras — APPROVED** [review]
LGTM :100:

**Comment by qmonnet** [comment]
@my-git9 This is not gonna work, one needs commit access to (re-)trigger the tests.
AKS test is currently broken, so expected to fail.

**Comment by my-git9** [comment]
> @my-git9 This is not gonna work, one needs to be a member of the Cilium organisation to (re-)trigger the tests. AKS test is currently broken, so expected to fail.

OK, thanks

**Comment by qmonnet** [comment]
AWS-CNI test is currently known to be broken, IPv6 smoke test as well (should be fixed by https://github.com/cilium/cilium/pull/19644/commits/953b1f9606bd9551546c5aabacd82ad323f53ac7 on master). `Cilium-PR-K8s-1.23-kernel-net-next` hits a flake. The changes are minor and I don't think there's a need to re-run these tests.

Tam and I both approved before the spelling-check change, so the reviews are in. I'm marking as ready-to-merge.

**Review by qmonnet — APPROVED** [review]

**Review by kaworu — APPROVED** [review]

**Pull request merged**